src/app/clusters/power-source-server/power-source-server.cpp

```diff
@@ -96,22 +142,24 @@ CHIP_ERROR PowerSourceServer::SetEndpointList(EndpointId powerSourceClusterEndpo
     // TODO: should check here that the power source cluster exists on the endpoint, but for now let's take the caller's word for it
 
     size_t idx = PowerSourceClusterEndpointIndex(powerSourceClusterEndpoint);
-    if (idx == std::numeric_limits<size_t>::max())
+    if (idx >= kNumSupportedEndpoints)
     {
         idx = NextEmptyIndex();
     }
-    if (idx == std::numeric_limits<size_t>::max())
+    if (idx >= kNumSupportedEndpoints)
     {
         return CHIP_ERROR_NO_MEMORY;
```

Comment:
This will likely not flow through to response IMStatus::ResourceExhausted unless done explicitly be caller in AddStatus

Reply:
It doesn't need to - this would be called internally, not by a client.